bin/commandTopicConsumer.py

```diff
@@ -0,0 +1,165 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Dump ksqlDB persistent queries to stdout.
+#
+# Usage:
+#  commandTopicConsumer.py -f <client-conf-file>
+#
+
+from confluent_kafka import KafkaError, KafkaException, version
+from confluent_kafka import Producer, Consumer
+import json
+import logging
+import argparse
+import uuid
+import sys
+import re
+
+
+class CommandRecord (object):
+    def __init__(self, stmt):
+        self.stmt = stmt
+
+    def __str__(self):
+        return "({})".format(self.stmt)
+
+    @classmethod
+    def deserialize(cls, binstr):
+        d = json.loads(binstr)
+        return CommandRecord(d['statement'])
+
+max_offset = -1001
+def consumer_latest_offsets(consumer, partitions):
+    global max_offset
+    for p in partitions:
+        high_water = consumer.get_watermark_offsets(p)[1]
+        if high_water >= max_offset:
+            max_offset = high_water
+
+class CommandConsumer(object):
+    def __init__(self, ksqlServiceId, conf):
+        self.consumer = Consumer(conf)
+        self.topic = '_confluent-ksql-{}_command_topic'.format(ksqlServiceId)
+
+    def consumer_run(self):
+        global max_offset
+        self.consumer.subscribe([self.topic], on_assign=consumer_latest_offsets)
+        #print("max offset = ", max_offset)
+        self.msg_cnt = 0
+        self.msg_err_cnt = 0
+        stmts = {}
+        try:
+            while True:
+                msg = self.consumer.poll(0.2)
+                if msg is None:
+                    continue
+
+                if msg.error() is not None:
+                    print("consumer: error: {}".format(msg.error()))
+                    self.consumer_err_cnt += 1
+                    continue
+
+                try:
+                    #print("Read msg with offset ", msg.offset())
+                    self.msg_cnt += 1
+                    record = CommandRecord.deserialize(msg.value())
+                    match = re.search(r'(?:create|drop) (?:stream|table) ([a-zA-z0-9-]+?)(:?\(|AS|\s|;)', record.stmt, re.I)
+                    if match:
+                        name = match.group(1).upper()
+                        if name == "KSQL_PROCESSING_LOG":
+                            continue
+                        if name not in stmts:
+                            stmts[name] = []
+                        stmts[name].append(record.stmt)
+                    else:
+                        match2 = re.search(r'(?:terminate) (?:ctas|csas)_(.+?)_', record.stmt, re.I)
```

Comment:
Maybe add a comment about what exactly you are assuming and extracting here? Is it just parsing the string version of every statement written to the command topic? Does this work with _all_ versions of KSQL?  Does it match all possible statements?

It may make sense to document that you are only

Reply:
I am parsing the `statement` part of the commands and it matches `CREATE table/stream`,  `DROP table/stream`, `CREATE TYPE`, `TERMINATE` and `INSERT INTO`. Are there other types of statements we want to match? 

I tried with 5.4, 5.3, 5.2 and 5.1 and the script works for all.

Reply:
What's the downside of outputting all statements?